### PR TITLE
Added the extended example for DebuggingDirective #21090

### PR DESCRIPTION
**Pull request by shiv4nsh**
Added the example and added the link to the documentation.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @shiv4nsh,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.knoldus.logging
+
+/**
```
Comment (drewhk) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala]: this header should go and instead the standard Lightbend Copyright header should be included.

**Comment by drewhk** [comment]
OK TO TEST

**Comment by shiv4nsh** [comment]
@ktoso , @drewhk : Can you please tell me what is the problem here. I have updated the code according to the comment. Please help me out here .

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import pl.project13.scala.rainbow.Rainbow
+import scala.concurrent.ExecutionContext.Implicits.global
+import Rainbow._
+
+import scala.concurrent.ExecutionContext
+
+
+//This example shows how to log client request and response time using DebuggingDirective
+
+class DebuggingDirectiveExtendedExample(implicit val system: ActorSystem,
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala]: ExampleSpec, and it should test it actually does what it's supposed to.

File: akka-docs/rst/scala/http/routing-dsl/directives/debugging-directives/logRequestResult.rst [hunk 3]
```diff
@@ -30,3 +30,13 @@ Example
 
 .. includecode2:: ../../../../code/docs/http/scaladsl/server/directives/DebuggingDirectivesExamplesSpec.scala
    :snippet: logRequestResult
+
+
+LongerExample : Logging the request and the response time of every request.
+-------------
```
Comment (ktoso) [inline akka-docs/rst/scala/http/routing-dsl/directives/debugging-directives/logRequestResult.rst]: underline should be as long as the text.
text is too long though.

```
Longer example
----------
```

And below explain what it is

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import akka.event.{LoggingAdapter, Logging}
+import akka.event.Logging.LogLevel
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.RouteResult.Complete
+import akka.stream.{Materializer, ActorMaterializer}
+import com.knoldus.logging.routes.KnoldusRoutingService
+import akka.http.scaladsl.server.directives.{LoggingMagnet, LogEntry, DebuggingDirectives}
+import pl.project13.scala.rainbow.Rainbow
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectiveExtendedExample.scala]: don't include rainbow in Akka please

**Comment by shiv4nsh** [comment]
@ktoso : I have updated the code as per the comments please review it :) Thanks

**Comment by shiv4nsh** [comment]
@ktoso : Now, I have updated the code as per the comments please review it :) Thanks

**Comment by shiv4nsh** [comment]
@ktoso : Can we merge this PR or else I have to rebase it

**Comment by shiv4nsh** [comment]
Added another PR with the code #21110

**Comment by ktoso** [comment]
Please don't be a bit patient. This PR is in a weird state, yes you have to rebase it - it should not include weird other commits (other than the one you want us to review and merge)

**Comment by shiv4nsh** [comment]
@ktoso : I have made another pull Request so with all the improvements as suggested by @johanandren  here: #21118

**Pull request closed without merging**